### PR TITLE
[DEV-1166] Add newsletter subscribe/unsubscribe button to profile's agreements page

### PR DESCRIPTION
**Pull request by marcobottaro**
#### List of Changes

Add newsletter subscribe/unsubscribe button to profile's agreements page

#### Motivation and Context

The authenticated user must be able to subscribe/unsubscribe from the newsletter by himself

#### How Has This Been Tested?

Manually

#### Screenshots (if appropriate):
image

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/nextjs-website/src/app/profile/agreements/page.tsx [hunk 1]
```diff
@@ -40,73 +88,127 @@ const Agreements = () => {
...
-      >
-        {t('newsletter.title')}
-      </Typography>
-      <Box
-        sx={{
-          display: 'flex',
-          flexDirection: { xs: 'column', md: 'row' },
-          gap: { xs: '10px', md: '100px' },
-        }}
+  // TODO: add dedicated loading and unauthorized pages
```
[the remark this answers is not in the document]
Comment (marcobottaro) [reply apps/nextjs-website/src/app/profile/agreements/page.tsx:91]: This is the loading task: https://pagopa.atlassian.net/browse/DEV-1159

This is the one dedicated to the "unauthorized" error page: https://pagopa.atlassian.net/browse/DEV-1167

File: apps/nextjs-website/src/app/profile/agreements/page.tsx [hunk 2]
```diff
@@ -2,21 +2,69 @@
...
   useTheme,
 } from '@mui/material';
 import Link from 'next/link';
-import React, { ReactNode } from 'react';
+import React, { ReactNode, useState } from 'react';
 import { ButtonNaked } from '@/editorialComponents/Footer/components/ButtonNaked';
+import { useUser } from '@/helpers/user.helper';
+import PageNotFound from '@/app/not-found';
+
+// TODO: Remove this code duplication and manage messages with a dedicated service
```
[the remark this answers is not in the document]
Comment (marcobottaro) [reply apps/nextjs-website/src/app/profile/agreements/page.tsx:21]: This is the task: https://pagopa.atlassian.net/browse/DEV-1047

File: apps/nextjs-website/src/app/profile/agreements/page.tsx [hunk 3]
```diff
@@ -40,73 +88,127 @@ const Agreements = () => {
...
+            fontSize: '14px',
+            color: palette.text.secondary,
+          }}
+        >
+          {privacyStatement}
+        </Typography>
+      </Stack>
+      <Snackbar
+        open={!!info}
+        autoHideDuration={4000}
```
Comment (MarcoPonchia) [inline apps/nextjs-website/src/app/profile/agreements/page.tsx:204]: Maybe is better to move this magic number to `apps/nextjs-website/src/config.ts`

Comment (marcobottaro) [reply apps/nextjs-website/src/app/profile/agreements/page.tsx:204]: This is the task: https://pagopa.atlassian.net/browse/DEV-1047

**Review by MarcoPonchia — APPROVED** [review]

**Review by kin0992 — APPROVED** [review]

**Pull request merged**